src/sidebar/util/groups.js

```diff
@@ -9,14 +12,15 @@ import serviceConfig from '../service-config';
  * explicitly disallowed in the service configuration of the
  * `settings` object.
  *
- * @param {object} settings
+ * @param {HostSettings} settings
  * @return {boolean}
  */
 function allowLeavingGroups(settings) {
-  const config = serviceConfig(settings) || {};
-  return typeof config.allowLeavingGroups === 'boolean'
-    ? config.allowLeavingGroups
-    : true;
+  const config = serviceConfig(settings);
+  if (!config) {
+    return true;
+  }
+  return !!config.allowLeavingGroups;
```

Comment:
This refactor avoids TS complaining that `config` might not have an `allowLeavingGroups` property, if it was initialized to `{}` (which is inferred to be an object with no properties).